### PR TITLE
Populate marketingABTests in the UDO

### PR DESCRIPTION
**Pull request by yonaik**
…sts expId and it's bucket value

### CODE REVIEW

**Comment by yonaik** [comment]
Initially I have considered few sample marketingABTests values to check our solution works. 
We need to get active and running marketingABTests value from Guillermo in this case.

File: src/library/simplified/extension/marketingABTests.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+var marketingABTests = "3659,14698,14739,13542,12102";
```
Comment (trungguyen) [inline src/library/simplified/extension/marketingABTests.js]: We don't need to know the active marketing AB Tests in this extension.

File: src/library/simplified/extension/marketingABTests.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+var marketingABTestsArray = marketingABTests.split(",");
```
Comment (trungguyen) [inline src/library/simplified/extension/marketingABTests.js]: Not needed.

File: src/library/simplified/extension/marketingABTests.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+var abTests = utag_data.context.tests;
+var marketingABTests = "";
+
+for (var abTest=0; abTest<abTests.length; abTest++) {
+    for (var marketingABTest=0; marketingABTest<marketingABTestsArray.length; marketingABTest++) {
```
Comment (trungguyen) [inline src/library/simplified/extension/marketingABTests.js]: We just need a single for loop to flatten the "tests" list, where you grab all the tests and their corresponding buckets and combined them all into one long string. Basically all you need to do for this extension is iterate through "tests" list and take each test and its bucket to concat them into the string format "testId.bucketId|testId.bucketId|testId.bucketId|...", and assign the string to a new udo variable, in this case marketingABTests, that you create in this extension. That's pretty much it, just make sure you set this extension to run prior to load rules. Then this string will be used in a load rule. For example, if we want a load rule to fire a tag based on a given marketing test ID defined (or given by) marketing team, let's say 12345, we will then have a load rule to check and see if "12345.1" exists in the string, marketingABTests, created by this extension. If the "marketingABTests" contains "12345.1".  I have outlined the steps for this under "Tealium" section of the confluence page.

**Review by trungguyen — CHANGES_REQUESTED** [review]

File: src/library/simplified/extension/marketingABTests.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+if (abTests != undefined) {
+    for (var abTest=0; abTest<abTests.length; abTest++) {
+            var experimentId = utag_data.context.tests[abTest].experimentId;
+            var bucketValue = utag_data.context.tests[abTest].bucketValue;
+            marketingABTests += experimentId + "." + bucketValue + "|";
+    }
+}
+
+marketingABTests = marketingABTests.substring(0,marketingABTests.length - 1);
```
Comment (trungnguyenexp) [inline src/library/simplified/extension/marketingABTests.js]: This is not necessary. one less instruction will help speed things up especially we don't want the page to be slow, but you can leave it as is if you want :).

File: src/library/simplified/extension/marketingABTests.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+var abTests = utag_data.context.tests;
```
Comment (trungnguyenexp) [inline src/library/simplified/extension/marketingABTests.js:1]: Please make sure an automation test case in Tag Management is created for to check for marketingABTests existent and it is not empty/null.

**Review by trungnguyenexp — COMMENTED** [review]
please see comments

**Pull request merged**